### PR TITLE
Issue 569/synonyms version2

### PR DESCRIPTION
**Pull request by ComLock**

### CODE REVIEW

File: src/main/resources/services/graphQL/synonym/addQuerySynonymsField.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+} from '../constants';
+import {querySynonyms} from './querySynonyms';
+
+
+//const and = storage.query.dsl.and;
+const bool = storage.query.dsl.bool;
+const fulltext = storage.query.dsl.fulltext;
+const ngram = storage.query.dsl.ngram;
+const or = storage.query.dsl.or;
+const stemmed = storage.query.dsl.stemmed;
```
Comment (alansemenov) [inline src/main/resources/services/graphQL/synonym/addQuerySynonymsField.ts:38]: Deconstruct?

File: src/main/resources/services/graphQL/synonym/addQuerySynonymsField.ts [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+				totalPages :number
+			};
+			result.localeToStemmingLanguage = localeToStemmingLanguage;
+			result.page = page;
+			result.start = start + 1;
+			result.end = Math.min(start + perPage, result.total);
+			result.totalPages = Math.ceil(result.total / perPage);
+
+			//log.debug('result:%s', toStr(result));
+			return result;
```
Comment (alansemenov) [inline src/main/resources/services/graphQL/synonym/addQuerySynonymsField.ts:255]: This should extracted and split up into several functions.

File: src/main/resources/tasks/init/model/16.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+export function model16({
+	progress,
+	writeConnection
+} :{
+	progress :Progress
+	writeConnection :WriteConnection
+}) {
+	progress.addItems(1).setInfo('Get all thesauri to lookup language:{from , to}...').report().logInfo();
+	const thesauri = getThesauri(writeConnection);
+	log.debug('thesauri:%s', toStr(thesauri));
```
Comment (alansemenov) [inline src/main/resources/tasks/init/model/16.ts:22]: Remove

File: src/main/resources/tasks/init/model/16.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+			fromLanguage,
+			toLanguage
+		} = thesauri[i];
+		progress.setInfo(`Processing thesaurus:${thesaurusName} fromLanguage:${fromLanguage} toLanguage:${toLanguage}...`).report().logInfo();
+		// Query for synonym nodes that has from and to
+		const synonyms = getSynonymsForThesaurusName({
+			thesaurusName,
+			writeConnection
+		});
+		log.debug('synonyms:%s', toStr(synonyms));
```
Comment (alansemenov) [inline src/main/resources/tasks/init/model/16.ts:38]: Remove

File: src/main/resources/tasks/init/model/16/modifySynonymNodeEditor.ts [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+		comment ?:string
+		enabled ?:boolean
+		synonyms :OneOrMore<SynonymItem>
+	}>
+	thesaurusReference :string
+	to ?:OneOrMore<string> // Optional since it's going to be removed
+};
+
+
+export function modifySynonymNodeEditor({
```
Comment (alansemenov) [inline src/main/resources/tasks/init/model/16/modifySynonymNodeEditor.ts:37]: Split into several sub-functions.

**Review by alansemenov — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/react/thesaurus/editSynonyms/EditSynonyms.tsx [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+								<Table.Cell><List>{synonymsObj.both.map((item, j) => <List.Item key={j}>{ReactHtmlParser(item)}</List.Item>)}</List></Table.Cell>
+								{thesaurusId ? null : <Table.Cell>{thesaurus}</Table.Cell>}
+								<Table.Cell>{_score}</Table.Cell>
+								<Table.Cell collapsing>
+									<DeleteSynonym
+										_id={_id}
+										from={synonymsObj.from}
+										afterClose={memoizedQuerySynonyms}
+										servicesBaseUrl={servicesBaseUrl}
+										to={synonymsObj.to}
```
[the remark this answers is not in the document]
Comment (ComLock) [reply src/main/resources/assets/react/thesaurus/editSynonyms/EditSynonyms.tsx]: Refactored

File: src/main/resources/assets/react/thesaurus/editSynonyms/useEditSynonymsState.ts [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+	});
+	const [thesauri, setThesauri] = React.useState(thesaurusName ? [thesaurusName] : []);
+	const [to, setTo] = React.useState('');
+
+	//──────────────────────────────────────────────────────────────────────────
+	// Functions
+	//──────────────────────────────────────────────────────────────────────────
+	// Since this function has no dependencies, there is no point in memoizing it.
+	const sortAfterColumnClick = (clickedColumn :string, currentColumn :string, currentDirection :string) => {
+		console.debug('sortAfterColumnClick(%s, %s, %s)', clickedColumn, currentColumn, currentDirection)
```
[the remark this answers is not in the document]
Comment (ComLock) [reply src/main/resources/assets/react/thesaurus/editSynonyms/useEditSynonymsState.ts]: fixed

File: src/main/resources/tasks/init/model/16.ts (hunk 3 above)
Comment (ComLock) [reply src/main/resources/tasks/init/model/16.ts:22]: Not yet, I want to look at it when I create the migration

**Pull request merged**